### PR TITLE
Us 1

### PR DESCRIPTION
**Pull request by ghost**
I did a text based merge on master into US-1. Now US-1 should not have any conflicts with master if the differing files in US-1 were to overwrite the corresponding files in master during a merge into master.

### CODE REVIEW

**Comment by ghost** [comment]
The reason github is complaining about merge problems is because the methods are declared in a different order between US-1 and master.

The git pull workflow works well if you can be certain that only "good" code goes into master, but right now master contains "bad" code, according to our code style guidelines. This means we can't merge US-1 into master automatically, even if we merge master into US-1 first, without putting the method declarations in US-1 into an arbitrary order that isn't aligned with our code style guidelines. Even if we were to do that, a formal review should deny the pull request.

In other words, if we want to pull the code in US-1 into master, we have to do it despite the conflicts and resolve them on the master, rather than on US-1.
